### DISCUSSION TITLE
How to delay container startup

### DISCUSSION THREAD

**Discussion by wormvortex** [photoprism/photoprism#1426] [Q&A] [answered]
When my VM running docker and in that PhotoPrism reboots I find that Photoprism cannot see my network drive and therefore thinks all the photos are missing. If I restart the PhotoPrism container all is fine however so I'm guessing that the container is starting before the network is ready.

What is the best way to make sure the network drive is ready before the container loads?

**Comment by rickysarraf** [comment]
It depends on what init service manager is running inside your VM.
Depending on the service manager, you'll need to appropriately add the
dependency.

For example, on systemd based systems, you'll have to put your docker
service to depend on your network share service.

s3nt fr0m a $martph0ne, excuse typ0s

**Reply by wormvortex** [reply]
So you're saying rather than delay the container I want to be looking at delaying the entire docker process from starting too early?

**Comment by rickysarraf** [comment]
Yes. Something like
https://unix.stackexchange.com/questions/246935/set-systemd-service-to-execute-after-fstab-mount

s3nt fr0m a $martph0ne, excuse typ0s

**Comment by lastzero** [comment, marked as answer]
Your files must be visible when PhotoPrism is running. Seems best to put a script in place that knows when your files were mounted properly, and delays startup for so long.

- There is no need to start PhotoPrism directly after your server or VM boots, although Docker offers this option. You can as well start PhotoPrism in a custom script via `docker-compose up -d` after your files were mounted.
- Alternatively you can mount a custom entrypoint script into the Docker image at `/entrypoint.sh`, ideally based on the existing one: https://github.com/photoprism/photoprism/blob/develop/docker/scripts/entrypoint.sh

**Comment by wormvortex** [comment]
Tried to delay the docker process starting but that doesn't seem to have helped. 

Still finding the VM boots, docker boots and then the photoprism container boots but must be happening before the VM shared folder is  ready as it can't see any folders until I restart the container.
